### PR TITLE
EZP-31795: Introduced proxy objects for content thumbnails

### PR DESCRIPTION
**Pull request by ViniTou**
| Question                                  | Answer
| ---------------------------------------- | ------------------
| **JIRA issue**                          | EZP-31795
| **Type**                                   | improvement
| **Target eZ Platform version** | `v3.2`
| **BC breaks**                          | no
| **Doc needed**                       | no

#### Checklist:
- [ ] Provided PR description.
- [ ] Tested the solution manually.
- [ ] Provided automated test coverage.
- [ ] Checked that target branch is set correctly (master for features, the oldest supported for bugs).
- [ ] Ran PHP CS Fixer for new PHP code (use `$ composer fix-cs`).
- [ ] Asked for a review (ping `@ezsystems/php-dev-team`).

### CODE REVIEW

**Review by Nattfarinn — APPROVED** [review]

**Review by lserwatka — APPROVED** [review]

File: eZ/Publish/Core/FieldType/Image/ImageThumbnailProxyStrategy.php [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+
+namespace eZ\Publish\Core\FieldType\Image;
+
+use eZ\Publish\API\Repository\Values\Content\Field;
+use eZ\Publish\API\Repository\Values\Content\Thumbnail;
+use eZ\Publish\Core\Repository\ProxyFactory\ProxyGeneratorInterface;
+use eZ\Publish\SPI\Repository\Strategy\ContentThumbnail\Field\FieldTypeBasedThumbnailStrategy;
+use ProxyManager\Proxy\LazyLoadingInterface;
+
+class ImageThumbnailProxyStrategy implements FieldTypeBasedThumbnailStrategy
```
Comment (Steveb-p) [inline eZ/Publish/Core/FieldType/Image/ImageThumbnailProxyStrategy.php:17]: Nitpick
```suggestion
final class ImageThumbnailProxyStrategy implements FieldTypeBasedThumbnailStrategy
```

**Review by Steveb-p — APPROVED** [review]

**Review by micszo — APPROVED** [review]
QA Approved with image, image asset sanities on Ibexa Commerce 3.2.3-dev and Ibexa Experience v3.2.3 with diff.

**Review by adamwojs — CHANGES_REQUESTED** [review]
+1 but you need add `\eZ\Publish\API\Repository\Values\Content\Thumbnail` to `\eZ\Bundle\EzPublishCoreBundle\Cache\Warmer\ProxyCacheWarmer::PROXY_CLASSES` See https://github.com/ezsystems/ezpublish-kernel/pull/2844 for more details.

**Comment by lserwatka** [comment]
@ViniTou we need to address it.

**Review by adamwojs — APPROVED** [review]

**Comment by lserwatka** [comment]
@ViniTou could you merge it up?

**Pull request merged**